src/rabbit_peer_discovery.erl

```diff
@@ -99,6 +103,56 @@ maybe_unregister() ->
   end.
 
 
+-spec maybe_inject_randomized_delay() -> ok.
+maybe_inject_randomized_delay() ->
+  Backend = backend(),
+  case Backend:supports_registration() of
```

Comment:
Oh, I thought I did that as part of https://github.com/rabbitmq/rabbitmq-server/issues/1143 but looks like that change was lost. Adding it now.